src/main/java/com/naijagov/naijagov/controller/NaijagovController.java

```diff
@@ -21,89 +21,116 @@
-                                      sortDirection.get(), sortBy.get());
-        //when navigation link is clicked
-        else if(form.getWardsPage()!=null) 
-            sortedPage = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), 
-                                        form.getWardsPage().getSort());
-        form.setWardsPage(wardService.listAllWards(sortedPage));
+  private PollingUnitService pollingUnitService;
+  private WardService wardService;
 
-        mv.setViewName("wards");
-        return mv;
+  @GetMapping("/wards")
+  public ModelAndView wards(
+      @PageableDefault(size = 10, sort = "fullCode", direction = Sort.Direction.ASC) Pageable pageable,
+      @ModelAttribute Form form, ModelAndView mv, @RequestParam("sortBy") Optional<String> sortBy,
+      @RequestParam("sortDirection") Optional<Sort.Direction> sortDirection) {
+    Pageable sortedPage = pageable;
+    // initial page load, no params
+    if (form.getWardsPage() == null) {
+      sortedPage = pageable;
     }
+    // when header sort is clicked
+    else if (sortBy.isPresent() && sortDirection.isPresent()) {
+      sortedPage = PageRequest.of(form.getWardsPage().getNumber(), form.getWardsPage().getSize(), sortDirection.get(),
+          sortBy.get());
+    }
+    // when navigation link is clicked
+    else if (form.getWardsPage() != null) {
+      sortedPage = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), form.getWardsPage().getSort());
+    }
+    form.setWardsPage(wardService.listAllWards(sortedPage));
+
+    mv.setViewName("wards");
+    return mv;
+  }
+
+  @GetMapping("/ward/{id}")
+  public ModelAndView displayWard(@PathVariable Integer id,
+      @PageableDefault(size = 10, sort = "fullCode", direction = Sort.Direction.ASC) Pageable pageable,
+      @ModelAttribute Form form, ModelAndView mv, @RequestParam("sortBy") Optional<String> sortBy,
+      @RequestParam("sortDirection") Optional<Sort.Direction> sortDirection) {
+
+    Optional<Ward> optionalWard = wardService.findWard(id);
+    ModelAndView modelAndView = new ModelAndView();
+    Ward wardObject = null;
+    if (optionalWard.isPresent()) {
+      // what's the significance of this?
+      wardObject = optionalWard.get();
+      modelAndView.addObject(wardObject);
 
-    @GetMapping("/ward/{id}")
-	public ModelAndView displayWard(@PathVariable Integer id) {
-        Optional<Ward> optionalWard = wardService.findWard(id);
-		ModelAndView modelAndView = new ModelAndView();
-        if(optionalWard.isPresent()) modelAndView.addObject(optionalWard.get());
-		modelAndView.setViewName("ward");
-		return modelAndView;		
-	}
+    }
+    // get pollingUnits for ward
+    Pageable sortedPage = pageable;
+    // initial page load, no params
+    if (form.getWardPage() == null) {
+      sortedPage = pageable;
+    }
+    // when header sort is clicked
+    else if (sortBy.isPresent() && sortDirection.isPresent())

```

Comment:
These conditions doesn't have curly-braces

Reply:
Fixed in [116ed54](https://github.com/bioye/naijagov/pull/68/commits/116ed54d06286edad2aa450cebca9f83e151a02f)